stock_barcodes_move_location/wizard/stock_barcodes_read_move_location.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2019 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _, fields, models
+from odoo.fields import first
+from odoo.addons import decimal_precision as dp
+
+
+class WizStockBarcodesReadMoveLocation(models.TransientModel):
+    _name = 'wiz.stock.barcodes.read.move.location'
+    _inherit = 'wiz.stock.barcodes.read'
+    _description = 'Wizard to read barcode on move location'
+
+    move_location_id = fields.Many2one(
+        comodel_name='wiz.stock.move.location',
+        string='Move Location',
+        readonly=True,
+    )
+    move_location_qty = fields.Float(
+        string='To Move quantities',
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True,
+    )
+
+    def name_get(self):
+        return [
+            (rec.id, '{} - {}'.format(
+                _('Barcode reader'),
+                self.env.user.name)) for rec in self]
+
+    def _prepare_move_location_line(self):
+
+        search_args = [
+            ('location_id', '=', self.move_location_id.origin_location_id.id),
+            ('product_id', '=', self.product_id.id),
+        ]
+        if self.lot_id:
+            search_args.append(('lot_id', '=', self.lot_id.id))
+        else:
+            search_args.append(('lot_id', '=', False))
+        res = self.env['stock.quant'].read_group(search_args, ['quantity'], [])
+        max_quantity = res[0]['quantity']
+        # Apply the putaway strategy
+        location_dest_id = \
+            self.move_location_id.destination_location_id.get_putaway_strategy(
+                self.product_id).id or self.location_dest_id.id
```

Comment:
An error appears when the destination location has no putaway strategy. I think it can be solved with:
`location_dest_id = \
            self.move_location_id.destination_location_id.get_putaway_strategy(
                self.product_id).id or self.location_id.id`

Reply:
This is now fixed. Thanks!